minesweeper.py

```diff
@@ -0,0 +1,196 @@
+        """Returns the number of mines surrounding given square."""
+        neighbors = []
+        mines_qty = 0
+
+        for i in range(-1, 2):
+            row_index = row + i
+            if row_index >= 0 and row_index < self.height:
+                if col > 0:
+                    neighbors.append((row_index, col - 1))
+                if col < self.width - 1:
+                    neighbors.append((row_index, col + 1))
+                neighbors.append((row_index, col))
+
+        for neighbor in neighbors:
+            if self.board[neighbor[0]][neighbor[1]]._mine:
+                mines_qty += 1
+
+        return mines_qty
+
+    def click_square(self, row, col):
+        """Updates clicked to True on game board for specified row and col."""
+
+        if self.board[row][col].clicked:
+            print self.board[row][col]
+            print 'You already clicked that square. Choose another.'
+        else:
+            self.board[row][col].clicked = True
+
+    def flag_square(self, row, col):
+        """Flags a square with a question mark."""
+
+        self.board[row][col].flagged = True
+
+    def is_mine(self, row, col):
+        """Returns True if specified row and col is a mine."""
+
+        return self.board[row][col]._mine
+
+    def display_board(self):
+        """Visually represent state of board."""
+
+        print ' ',
+        for n in range(self.width):
+            print ' {} '.format(n),
+        print ''
+        for i, row in enumerate(self.board):
+            print i,
+            for col in self.board[row]:
+                if self.board[row][col].clicked:
+                    if self.board[row][col]._mine:
+                        print '[*]',
+                    else:
+                        mines = self._count_touching_mines(row, col)
+                        print ' {} '.format(mines),
+                elif self.board[row][col].flagged:
+                    print '[?]',
+                else:
+                    print '[ ]',
+            print '\n'
+
+
+class Minesweeper(object):
+
+    def __init__(self):
+        self.keep_playing = True
+        self.current_game = Board()
+        self._click_qty = 0
+        self.start_game()
+
+    def start_game(self):
+        """Initiates minesweeper game."""
+
+        print 'Welcome to Minesweeper!'
+        response = raw_input('Current board is 10x10 with 20 mines. Press enter to begin, or type CUSTOM to modify board. > ')
+
+        if response.lower() == 'custom':
+            rows = raw_input('How many rows? > ')
+            cols = raw_input('How many columns? > ')
+            mines = raw_input('How many mines? > ')
+
+            try:
+                self.current_game = Board(int(mines), int(rows), int(cols))
+            except:
+                print 'Not valid choices. Try again.'
+
+        print '\nType SHOW BOARD at any time to see current status.\n'
+        self.play_game()
+
+    def is_win(self):
```

Comment:
Seems unexpected that `is_win()` mutates state -- is there a better method name on this?